src/reanimated2/Colors.js

```diff
@@ -386,6 +386,42 @@ function normalizeColor(color) {
   return null;
 }
 
+function red(color) {
+  'worklet';
+  return (color >> 24) & 255;
+}
+
+function green(color) {
+  'worklet';
+  return (color >> 16) & 255;
+}
+
+function blue(color) {
+  'worklet';
+  return (color >> 8) & 255;
+}
+
+function alpha(color) {
+  'worklet';
+  return (color & 255) / 255;
+}
+
+export function destructColor(color) {
+  'worklet';
+
+  const normalizedColor = normalizeColor(color);
+  if (normalizedColor === undefined || normalizedColor === null) {
+    return undefined;
+  }
+
+  const r = red(normalizedColor);
+  const g = green(normalizedColor);
+  const b = blue(normalizedColor);
+  const a = alpha(normalizedColor);
```

Comment:
why are all those calculations wrapped into separate functions?

Reply:
> why are all those calculations wrapped into separate functions?

For better code reusability and ability to extract and use color channel later, when more code for working with colors will be added.